cargo-auditable/src/object_file.rs

```diff
@@ -197,8 +200,9 @@ fn riscv_features(target_triple: &str) -> String {
     // Most but not all riscv targets declare target features.
     // A notable exception is `riscv64-linux-android`.
     // We assume that all Linux-capable targets are -gc.
-    if target_triple.contains("linux") {
-        extensions.push_str("imadfc");
+    match info["target_os"].as_str() {
+        "linux" | "android" => extensions.push_str("imadfc"),
+        _ => (),
```

Comment:
With the exception of android, all riscv target triples contain the set of enabled extensions that matter for the ABI and a bunch more right after the riscv32/riscv64 part of the target triple name with `g` being an alias for `imafd`. So for example `riscv64imac-unknown-none-elf` has `imac` as features, while `riscv64gc-unknown-freebsd` has `imafdc`.